changelog/11904.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+api: extracting Unix domain socket handling to separate function
```

Comment:
I would describe the change from the user perspective, i.e. what's the behavioural change?  If there's none, probably no changelog is needed.  In your case, I would say something like "support UNIX domain sockets in Client.SetAddress".